File: src/main/java/org/featx/jvmon/entity/AbstractUpdate.java
package org.featx.jvmon.entity;

import lombok.Data;
import lombok.EqualsAndHashCode;
import java.time.LocalDateTime;

/**
 * @author Excepts
 */
@Data
@EqualsAndHashCode(callSuper = true)
public abstract class AbstractUpdate<I> extends AbstractRecord<I> implements Update<I> {

    private LocalDateTime updatedAt;
}

File: src/main/kotlin/org/featx/jvmon/entity/AbstractUpdate.kt
package org.featx.jvmon.entity

import lombok.Data
import lombok.EqualsAndHashCode

import java.time.LocalDateTime

@Data
@EqualsAndHashCode(callSuper = true)
abstract class AbstractUpdate<I> : AbstractRecord<I>(), Update<I> {

    override var updatedAt: LocalDateTime? = null
        set(updatedAt) {
            field = this.updatedAt
        }
}
